tests/test_system.py

```diff
@@ -0,0 +1,210 @@
+                                                'web/4.jpg',
+                                                ('http://unt.edu',))
+        expected = (None, None)
+        assert (path, location) == expected
+
+
+class TestGetFilePath():
+
+    def test_get_file_path(self):
+        """Test correct creation of file path."""
+        file_path = system.get_file_path('metapth', 'file://web/01_tif/4.jpg')
+        expected_path = '/me/ta/pt/h/metapth/web/01_tif/4.jpg'
+        assert file_path == expected_path
+
+    def test_get_file_path_raises_exception(self):
+        """Test invalid file name raises SystemMethodsException."""
+        with pytest.raises(system.SystemMethodsException):
+            system.get_file_path('metapthx', '/web/01_tif/4.jpg')
+
+
+class TestGetCompleteFilePath():
+
+    def test_get_complete_filepath(self):
+        """Test correct generation of the complete file path."""
+        complete_path = system.get_complete_filepath('metapthx',
+                                                     'file://web/4.jpg',
+                                                     'http://unt.edu/disk2/')
+        expected_path = 'http://unt.edu/disk2/me/ta/pt/hx/metapthx/web/4.jpg'
+        assert complete_path == expected_path
+
+
+class TestOpenSystemFile():
+
+    @mock.patch('urllib2.urlopen')
+    def test_open_system_file_http(self, mocked_urlopen):
+        """Test return value comes from urlopen call."""
+        mocked_urlopen.return_value = expected = 'file'
+        file_obj = system.open_system_file('http://example.com/pth/f.jpg')
+        assert file_obj == expected
+
+    @mock.patch('urllib2.urlopen')
+    @mock.patch('aubreylib.system.get_other_system')
+    def test_open_system_file_http_other_system(self,
+                                                mocked_other_system,
+                                                mocked_urlopen):
+        """Test return value comes from get_other_system call."""
+        mocked_urlopen.side_effect = Exception()
+        mocked_other_system.return_value = expected = 'file'
+        file_obj = system.open_system_file('http://example.com/pth/f.jpg')
+        assert file_obj == expected
+
+    @mock.patch('__builtin__.open')
+    def test_open_system_file_local(self, mocked_open):
+        """Test return value comes from local location."""
+        mocked_open.return_value = expected = 'file'
+        file_obj = system.open_system_file('/pth/f.jpg')
+        assert file_obj == expected
+
+
+class TestOpenArgsSystemFile():
+
+    @mock.patch('urllib2.urlopen')
+    def test_open_args_system_file_returns_file(self, mocked_urlopen):
+        """Test a valid URL results in a returned object."""
+        mocked_urlopen.return_value = expected = 'file'
+        url_with_args = 'http://example.com/pth/f.jpg?start=123'
+        file_obj = system.open_args_system_file(url_with_args)
+        assert file_obj == expected
+        mocked_urlopen.assert_called_once_with(url_with_args)
```

Comment:
I wanted to verify not only that it was called, but that it was called with the specific argument since the url that `open_args_system_file` is called with is not actually the same variable used as the argument of the `urlopen` call (though the value may or may not be the same).